### ISSUE TITLE
Old jobs not being removed from cron list

### ISSUE THREAD

**Issue by lloydwatkin** [sidekiq-cron/sidekiq-cron#504]
We've removed some jobs from our schedule, but they still seem to be being called and not removed from the list.

* rails (~>7.2)
* sidekiq (7.3.2)
* sidekiq-cron (1.12.0)

Is there a way to force reset the list?

**Comment by markets** [comment]
Hello @lloydwatkin 👋🏼  

What method are you using https://github.com/sidekiq-cron/sidekiq-cron#loading-jobs-from-schedule-file? If you are using the "automatic schedule loader", jobs are clean-up automatically. If you are doing some manual work, don't forget to use the bang suffixed method `load_from_hash!`. 

More info: https://github.com/sidekiq-cron/sidekiq-cron/pull/431

**Comment by lloydwatkin** [comment]
@markets thanks for the reply. This is what we're using and it hasn't changed in a long time...

```ruby
  schedule_file = "#{Rails.root}/config/job-schedule.yml"

  if File.exist?(schedule_file) && Sidekiq.server?
    Sidekiq::Cron::Job.load_from_hash! YAML.load_file(schedule_file)
  end
```

I notice in the docs it has `source: 'schedule'` is that maybe the issue?

**Comment by lloydwatkin** [comment]
[Looks like it](https://github.com/sidekiq-cron/sidekiq-cron/pull/431/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R54), I'll make the change and reopen if that doesn't fix. Thanks!

**Comment by markets** [comment]
Yes, that seems the cause! Please review this thread for further information #431. But basically we need to know if a job is created from the schedule file or dynamically.

Going to close it for now, but please reopen if that doesn't fix your issue.
